clustered/common/src/main/java/org/ehcache/clustered/common/messages/ServerStoreOpMessage.java

```diff
@@ -78,16 +97,48 @@ public long getKey() {
 
   public abstract ServerStoreOp operation();
 
+  public byte[] encode() {
+    ByteBuffer encodedMsg = ByteBuffer.allocate(MSG_TYPE_SIZE + STORE_OP_CODE_SIZE + CACHE_ID_LEN_SIZE + KEY_SIZE + getCacheId().length());
+    encodedMsg.put(EhcacheEntityMessage.Type.SERVER_STORE_OP.getOpCode());
+    encodedMsg.put(this.operation().getStoreOpCode());
+    encodedMsg.putInt(this.cacheId.length());
+    encodedMsg.put(this.cacheId.getBytes(UTF_8));
```

Comment:
The above two lines assume that UTF-8 will encode all characters as single bytes which is only true for ASCII characters. You should rather do:

```
byte[] cacheIdBytes = cacheId.getBytes(UTF_8);
encodedMsg.putInt(cacheIdBytes.length);
encodedMsg.put(cacheIdBytes);
```